subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptions.java

```diff
@@ -408,4 +409,23 @@ public void applyTo(StartParameterInternal settings, Origin origin) {
             settings.setExportKeys(true);
         }
     }
+
+    public static class InstantExecutionOption extends BooleanBuildOption<StartParameterInternal> {
+
+        public static final String PROPERTY_NAME = "org.gradle.unsafe.instant-execution";
+        public static final String LONG_OPTION = "instant-execution";
+
+        public InstantExecutionOption() {
+            super(PROPERTY_NAME, BooleanCommandLineOptionConfiguration.create(
+                LONG_OPTION,
+                "Enables instant execution. Gradle will try to reuse the build configuration from previous builds.",
```

Comment:
perhaps include something like "[incubating]" or "[experimental]" in the description.

Reply:
Good point. `BooleanCommandLineOptionConfiguration` has a `.incubating()` function that do that uniformly.